### PR TITLE
Stop importing from nuxt.config.js

### LINKED ISSUE #524: Stop importing `nuxt.config.js` into the app runtime
## Description

Currently we import `nuxt.config.js` into the app's runtime, mainly to reference the `dev` property of the config object.

This is fine in Nuxt 2 but will not be supported in Nuxt 3 because Nuxt 3's configuration file requires you to import from Nuxt 3's build-only toolkit. Trying to import this toolkit during runtime will cause webpack to crash.

Likewise, it also bundles our entire configuration file into the frontend bundle, which isn't necessary and could be unexpected, especially if secrets get interpolated into that file by Webpack's environment variable resolution.

## Reproduction
1. Visit https://wordpress.org/openverse and open the JavaScript debugger. Inspect the files coming from search.openverse.engineering and in each file search for a pattern that would match the config file (like `dev:`) and you'll be able to find the configuration file in total bundled into the frontend code.

Captura de Tela 2021-12-14 às 12 03 48

This is not currently a vulnerability but it could unexpectedly lead to one via an unintentionally leaked secret.

## Expectation
We should move the `dev` variable into it's own module (`dev.js`) and import _that_ into the config file and in the runtime. This would allow us to avoid duplicating the logic around the app while still preventing the import of the config file.

## Additional Context
https://github.com/nuxt/framework/issues/1320

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #524 by @sarayourfriend 

## Description

Stops importing from `nuxt.config.js` during the runtime. See related issue for the rationale behind this.

## Testing Instructions

Run `pnpm dev` and `pnpm storybook` and verify that they work as expected. There should be no changes to the behavior of the application.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/locales/scripts/wp-locales.json [hunk 1]
```diff
@@ -467,7 +467,7 @@
     "facebookLocale": "el_GR",
     "wpLocale": "el",
     "code": "el",
-    "translated": 9
+    "translated": 28
```
Comment (AetherUnbound) [inline src/locales/scripts/wp-locales.json:470]: This looks like the change from #545, should this PR be pointed at that branch?

Comment (sarayourfriend) [reply src/locales/scripts/wp-locales.json:470]: Whoops, I didn't mean to include that file! I'll remove it from the commit.

File: src/composables/use-i18n-sync.js [hunk 2]
```diff
@@ -130,7 +129,6 @@ export default function useI18nSync() {
   onMounted(() => {
     window.addEventListener('message', localeMsgHandler)
     sendWindowMessage({
```
Comment (obulat) [inline src/composables/use-i18n-sync.js:131]: Can we import `dev` from the `utils/dev` here, too? It's usually helpful to see the message exchange when debugging the iframe.

**Review by obulat — APPROVED** [review]
LGTM

File: src/utils/env.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Use relative import to resolve during nuxt runtime
+import stringToBoolean from './string-to-boolean'
+
+/**
+ * Default environment variables are set on this key. Defaults are fallbacks to existing env vars.
+ * All boolean values should be designed to be false by default.
+ */
+export const env = {
```
Comment (dhruvkb) [inline src/utils/env.js:8]: I wonder if there is a performance boost to using `Object.freeze` on this object literal?

**Review by dhruvkb — APPROVED** [review]
LGTM, this is great!

File: src/utils/env.js (hunk 3 above)
Comment (sarayourfriend) [reply src/utils/env.js:8]: Maybe? I doubt it's a measurable difference though.

**Pull request merged**